src/init.cpp

```diff
@@ -893,21 +892,30 @@ bool AppInitParameterInteraction()
     }
 
     // Make sure enough file descriptors are available
-    int nBind = std::max(
+    const unsigned int nBind = std::max(
                 (mapMultiArgs.count("-bind") ? mapMultiArgs.at("-bind").size() : 0) +
                 (mapMultiArgs.count("-whitebind") ? mapMultiArgs.at("-whitebind").size() : 0), size_t(1));
-    nUserMaxConnections = GetArg("-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS);
-    nMaxConnections = std::max(nUserMaxConnections, 0);
+
+    // Verify the number of connections, then set nUserMax
+    const int nUserMaxConnections = GetArg("-maxconnections", DEFAULT_MAX_PEER_CONNECTIONS);
+
+    if (nUserMaxConnections < 1)
+        return InitError(strprintf(_("Max connections %d too low."), nUserMaxConnections));
```

Comment:
This changes the behavior of node startup (hard abort instead of softly & silently applying a 'correction') so operators may run into some unexpected results when upgrading from a previous version to this. 

Other than that, concept ACK.

Reply:
Oh good point, let me downgrade that to a debug message. I want to keep *some* behavior around this, because it makes the control flow and types easier after it, but let's not change node startup behavior for something like this.